L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc

```diff
@@ -307,9 +339,10 @@ void L1FPGATrackProducer::beginRun(const edm::Run& run, const edm::EventSetup& i
   settings.setBfield(mMagneticFieldStrength);
 
   setup_ = iSetup.getData(esGetToken_);
+  channelAssignment_ = const_cast<ChannelAssignment*>(&iSetup.getData(esGetTokenChannelAssignment_));
```

Comment:
Why is const_cast needed here?

Reply:
in order to work with ChannelAssignment* instead of const ChannelAssignment*

Reply:
This code compiles fine if const_cast is removed, and "ChannelAssignment*" is replaced with "const ChannelAssignment*" everywhere.

Reply:
I try to avoid const data member

Reply:
Official code reviewers have made clear in past that we should avoid const_cast unless no other solution. Also, our code doesn't  do a "move" operation on TrackletEventProcessor, so it's not essential that its data members be non-const. (And indeed it already has "const Settings * settings_", which would prevent this). Please remove const_cast.
P.S. The official CMSSW style guide http://cms-sw.github.io/cms_coding_rules.html says "Make const all methods, data members ... that do not need to be non-const".

Reply:
done